app/controllers/submissions_controller.rb

```diff
@@ -800,6 +803,7 @@ def get_all_file_data(revision, grouping, path)
     return [] unless revision.path_exists?(full_path)
 
     anonymize = current_role.ta? && grouping.assignment.anonymize_groups
+    url_allowed = grouping.assignment.url_submit
```

Comment:
This is nitpicky but....

You have 3 variables that are named differently but all have the same semantics: `url_allowed`, `url_enabled`, `url_submit`. Pick one and stick with it please